deploy/fermyon-developer.nomad

```diff
@@ -99,9 +99,9 @@ job "fermyon-developer" {
       driver = "exec"
 
       artifact {
-        source = "https://github.com/fermyon/spin/releases/download/v0.8.0/spin-v0.8.0-linux-amd64.tar.gz"
+        source = "https://github.com/fermyon/spin/releases/download/v0.10.0/spin-v0.10.0-linux-amd64.tar.gz"
```

Comment:
10.1 for security patch?

Reply:
Was planning to do that tomorrow across the board for `fermyon.com`, `finicky-whiskers`, and `developer`. FWIW IIUC the CVE isn't really a factor here because we aren't actually running guest code; however i do plan to update to `v0.10.1`